scraper.rb

```diff
@@ -1,95 +1,35 @@
 #!/bin/env ruby
 # encoding: utf-8
 # frozen_string_literal: true
-
+require 'json'
 require 'pry'
+require 'require_all'
 require 'scraped'
 require 'scraperwiki'
 
 # require 'open-uri/cached'
 # OpenURI::Cache.cache_path = '.cache'
 require 'scraped_page_archive/open-uri'
 
-def noko_for(url)
-  Nokogiri::HTML(open(url).read)
-end
-
-def party_info_for(img)
-  return ['Partido Revolucionario Institucional', 'PRI'] if img == 'images/pri01.png'
-  return ['Partido Acción Nacional', 'PAN'] if img == 'images/pan.png'
-  return ['Partido de la Revolución Democrática', 'PRD'] if img == 'images/prd01.png'
-  return ['Partido Verde Ecologista', 'PVE'] if img == 'images/logvrd.jpg'
-  return ['Movimiento Ciudadano', 'MC'] if img == 'images/logo_movimiento_ciudadano.png'
-  return ['Partido del Trabajo', 'PT'] if img == 'images/logpt.jpg'
-  return ['Partido Nueva Alianza', 'PANAL'] if img == 'images/panal.gif'
-  return ['Movimiento Regeneración Nacional', 'MORENA'] if img == 'images/LogoMorena.jpg'
-  return ['Encuentro Social', 'PES'] if img == 'images/encuentro.png'
-  return ['Sin Partido', '_IND'] if img == 'images/logo_SP.jpg'
-  return ['Sin Partido', '_IND'] if img == 'images/independiente.png'
-  raise "Unknown party: #{img}"
-end
+require_rel 'lib'
 
-def ocd(entidad, d_c_text)
-  d_c, d_c_no = d_c_text.split(/\.\s+/, 2)
-  dc_expanded = {
-    'Circ' => 'Circunscripción',
-    'Dtto' => 'Distrito',
-  }
-  area = '%s %s' % [entidad, d_c_no]
-  area_id = 'ocd-division/country:mx/entidad:%s/%s:%s' % [entidad, dc_expanded[d_c], d_c_no].map do |str|
-    str.downcase.gsub(/[[:space:]]+/, '-')
-  end
-  [area, area_id]
+def scrape(h)
+  url, klass = h.to_a.first
+  klass.new(response: Scraped::Request.new(url: url).response)
 end
 
-def scrape_list(url)
-  noko = noko_for(url)
-  table = noko.xpath('//table[.//td[contains(.,"Entidad")]]').last
-
-  party = party_id = nil
-  table.xpath('.//tr').each_with_index do |tr, i|
-    tds = tr.css('td')
-    if tds.size == 1
-      if img = tr.css('img/@src').text
-        next if img.to_s.empty?
-        next if img == 'images/h_line.gif'
-        next if img == 'images/lin_por_gp.jpg'
-        party, party_id = party_info_for(img)
-        warn party.to_s
-      end
-      next
+def scrape_list(term, url)
+  (scrape url => MembersPage).party_groupings.flat_map do |party|
+    party.members.map do |mem|
+      {
+        party:    party.name,
+        party_id: party.id,
```

Comment:
I'd been checking against a CSV generated from the SQL. I've now started using `puts data.reject { |k, v| v.to_s.empty? }.sort_by { |k, v| k }.to_h` instead.